code/services/LDAPService.php

```diff
@@ -413,10 +414,50 @@ public function updateMemberFromLDAP($member, $data = null) {
 				Group::get()->byId($groupRecord['GroupID'])->Members()->remove($member);
 			}
 		}
-
 		// This will throw an exception if there are two distinct GUIDs with the same email address.
 		// We are happy with a raw 500 here at this stage.
 		$member->write();
 	}
 
+
+	/**
+	 * Change a members password on the AD. Works with ActiveDirectory compatible services that saves the
+	 * password in a `unicodePwd` field.
+	 *
+	 * Ensure that the LDAP bind:ed user can change passwords and that the connection is secure.
+	 *
+	 * @param Member $member
+	 * @param string $password
+	 * @return ValidationResult
+	 * @throws Exception
+	 */
+	public function setPassword(Member $member, $password) {
+		// @todo(stig): _t() all the error messages
+		$validationResult = ValidationResult::create(true);
+		if (!$member->GUID) {
+			SS_Log::log(sprintf('Cannot update Member ID %s, GUID not set', $member->ID), SS_Log::WARN);
+			$validationResult->error(_t('LDAPAuthenticator.PROBLEMFINDINGDATA', 'There was a problem retrieving your user data'));
+			return $validationResult;
+		}
+
+		$userData = $this->getUserByGUID($member->GUID);
+		if(empty($userData['distinguishedname'])) {
+			$validationResult->error(_t('LDAPAuthenticator.PROBLEMFINDINGDATA', 'There was a problem retrieving your user data'));
+		}
+		$properties["unicodePwd"] = iconv("UTF-8", "UTF-16LE", '"' . $password . '"');
```

Comment:
It seems like running the escapeValue totally messes up the password and it's impossible to login. Doing some light reading around the subject it seems like only queries needs to be escaped.